nifi-nar-bundles/nifi-zendesk-bundle/nifi-zendesk-processors/src/main/java/org/apache/nifi/processors/zendesk/GetZendesk.java

```diff
@@ -0,0 +1,375 @@
+        List<ValidationResult> results = new ArrayList<>(2);
+
+        String password = validationContext.getProperty(ZENDESK_PASSWORD).evaluateAttributeExpressions().getValue();
+        String token = validationContext.getProperty(ZENDESK_TOKEN).evaluateAttributeExpressions().getValue();
+        if (isNotBlank(password) && isNotBlank(token)) {
+            results.add(new ValidationResult.Builder()
+                .subject(ZENDESK_PASSWORD_NAME)
+                .valid(false)
+                .explanation("Either password or token should be set.")
+                .build());
+        }
+
+        ZendeskExportMethod exportMethod = ZendeskExportMethod.forName(validationContext.getProperty(ZENDESK_EXPORT_METHOD).getValue());
+        ZendeskResource zendeskResource = ZendeskResource.forName(validationContext.getProperty(ZENDESK_RESOURCE).getValue());
+        if (!zendeskResource.supportsExportMethod(exportMethod)) {
+            results.add(new ValidationResult.Builder()
+                .subject(ZENDESK_EXPORT_METHOD_NAME)
+                .valid(false)
+                .explanation("Not supported export method for resource.")
+                .build());
+        }
+
+        return results;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        webClientServiceProvider = context.getProperty(WEB_CLIENT_SERVICE_PROVIDER).asControllerService(WebClientServiceProvider.class);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) {
+        ZendeskResource zendeskResource = ZendeskResource.forName(context.getProperty(ZENDESK_RESOURCE).getValue());
+        ZendeskExportMethod exportMethod = ZendeskExportMethod.forName(context.getProperty(ZENDESK_EXPORT_METHOD).getValue());
+
+        URI uri = createUri(context, zendeskResource, exportMethod);
+        HttpResponseEntity response = performQuery(context, uri);
+
+        if (response.statusCode() == OK.getCode()) {
+            AtomicInteger resultCount = new AtomicInteger(0);
+            FlowFile createdFlowFile = session.write(
+                session.create(),
+                httpResponseParser(context, response, zendeskResource, exportMethod, resultCount));
+            int recordCount = resultCount.get();
+            if (recordCount > 0) {
+                FlowFile updatedFlowFile = session.putAttribute(createdFlowFile, RECORD_COUNT_ATTRIBUTE_NAME, Integer.toString(recordCount));
+                session.getProvenanceReporter().receive(updatedFlowFile, uri.toString());
+                session.transfer(updatedFlowFile, REL_SUCCESS);
+            } else {
+                session.remove(createdFlowFile);
+            }
+        } else if (response.statusCode() == HTTP_TOO_MANY_REQUESTS) {
+            context.yield();
+            throw new ProcessException(format("Rate limit exceeded, yielding before retrying request."));
```

Comment:
What's the goal of throwing here?
ProcessSession is going to catch this exception, roll back the session and make the processor yield anyway. But there is no flowfile related operation in this scenario to roll back, so other than logging, nothing is going to happen here.

Reply:
Thanks for the highlight. There was no particular intention with the throw. Will remove the throw clause, and log the error instead.